R/plot_fns.R

```diff
@@ -7109,6 +7144,125 @@ statres_volcano_plot <-
+#' @param text_size An integer specifying the size of the text (number of
+#'   non-missing values) within the plot. The default is 3.
+#' @param display_count logical value. Indicates whether the non-missing counts will
+#'   be displayed on the bar plot. The default is TRUE.
+#' @param x_lab character string specifying the x-axis label.
+#' @param y_lab character string specifying the y-axis label.
+#' @param title_lab character string specifying the plot title.
+#' @param legend_lab character string specifying the legend title.
+#' @param only_significant logical specifying whether only significant fold changes
+#'    are included in the histogram or all fold changes
+#'
+#' @return `ggplot` object - histogram
+#'
+#' @keywords internal
+#'
+statres_histogram <- 
+  function(x,
+           fc_colors = c("blue","darkgray"),
+           text_size,
+           x_lab,
+           y_lab,
+           title_lab,
+           legend_lab,
+           free_y_axis,
+           only_significant) {
+    # If free_y_axis is true then each bar plot can have its own y axis range.
+    if (free_y_axis) {
+      da_scales <- "free"
+    } else {
+      da_scales <- NULL
+    }
+    
+    only_sig <- if (is.null(only_significant))
+      TRUE else
+        only_significant
+    
+    # Farm boy, do all the tedious label crap. As you wish.
+    the_x_label <- if (is.null(x_lab))
+      "log2 Fold Change" else
+        x_lab
+    the_y_label <- if (is.null(y_lab))
+      "Count of Biomolecules" else
+        y_lab
+    the_title_label <- if (is.null(title_lab))
+      "Distribution of log2 Fold Changes" else
+        title_lab
+    the_legend_label <- if (is.null(legend_lab))
+      "Fold Change Significance" else
+        legend_lab
+    
+    
+    # make the data long by anova comparisons, p-values, and flag
+    x_long <- x %>%
+      tidyr::pivot_longer(dplyr::starts_with("Fold_change_"),
+                          names_to = "Comparison",
+                          values_to = "Fold_change") %>%
+      dplyr::mutate(Comparison = stringr::str_remove(Comparison,"Fold_change_")) %>%
+      tidyr::pivot_longer(dplyr::starts_with("P_value_A"),
+                          names_to = "Comparison2",
+                          values_to = "P_value_A") %>%
+      dplyr::mutate(Comparison2 = stringr::str_remove(Comparison2, "P_value_A_")) %>%
+      dplyr::filter(Comparison == Comparison2) %>%
+      dplyr::select(-Comparison2) %>%
+      tidyr::pivot_longer(dplyr::starts_with("Flag_A"),
+                          names_to = "Comparison3",
+                          values_to = "Flag_A") %>%
+      dplyr::mutate(Comparison3 = stringr::str_remove(Comparison3, "Flag_A_")) %>%
+      dplyr::filter(Comparison == Comparison3) %>%
+      dplyr::select(-Comparison3) %>%
+      dplyr::mutate(Significant = ifelse(Flag_A == 0, "Not Significant","Significant"))
+    
+    
+    
+    if(only_sig == TRUE){
+      p <- x_long %>%
+        dplyr::filter(P_value_A < 0.05) %>%
```

Comment:
This ought to be informed by the designated significance threshold in their statRes object, not 0.05

Can extract with attr(x, "pval_thresh")